### PR TITLE
Document Longitude wrap_angle angle-like

### PR DESCRIPTION
**Pull request by nstarman**
### Description

Document the wrap_angle argument of Longitude angle-like (accept strings) ~and deprecate passing Angle subclasses~.

Followup to: #11118 
@adrn @mhvk @ayshih 

tag: no-changelog-needed

### CODE REVIEW

**Comment by ayshih** [comment]
I see some of the discussion here (https://github.com/astropy/astropy/pull/11118#discussion_r606665399), but  blocking all possible subclasses seems severe and contrary to the spirit of class inheritance.  If the concern is that the specific subclass of `Latitude` doesn't make sense, then check for that.

Plus, it's peculiar to say that "angle-like" input is allowed, but then only allow `Angle`.  If that's truly the only allowed class, then just name that specific input type.

**Comment by ayshih** [comment]
(It's a little funky, but I think that `Longitude` should be valid input for the wrap angle)

**Comment by nstarman** [comment]
@ayshih Thanks for the input!

> I see some of the discussion here (#11118 (comment)), but blocking all possible subclasses seems severe and contrary to the spirit of class inheritance. If the concern is that the specific subclass of Latitude doesn't make sense, then check for that...
(It's a little funky, but I think that Longitude should be valid input for the wrap angle)

I'm fine with prohibiting specific classes over a blanket ban. Latitude would certainly be on the list. Longitude feels very odd. I suppose it's valid. So long as it's documented, we don't need to hand-hold.

> Plus, it's peculiar to say that "angle-like" input is allowed, but then only allow Angle. If that's truly the only allowed class, then just name that specific input type.

Agreed. The docs should be updated to something like "Angle, str, None, number". or perhaps "angle-like, no Angle subclass instance"". If we keep the blanket ban.

**Comment by nstarman** [comment]
Ok. undeprecated angle subclasses. and rebased. should be good.
Also cleaned up the imports a little in ``test_angles``.

File: astropy/coordinates/angles.py [hunk 1]
```diff
@@ -664,7 +664,8 @@ def __new__(cls, angle, unit=None, wrap_angle=None, **kwargs):
         self = super().__new__(cls, angle, unit=unit, **kwargs)
         if wrap_angle is None:
             wrap_angle = getattr(angle, 'wrap_angle', self._default_wrap_angle)
-        self.wrap_angle = wrap_angle
+
+        self.wrap_angle = Angle(wrap_angle, copy=False)
```
Comment (mhvk) [inline astropy/coordinates/angles.py:668]: I'm confused. Why is this necessary? the `wrap_angle` setter already does this. Don't the additional tests pass without this?

Comment (nstarman) [reply astropy/coordinates/angles.py:668]: Good catch! I missed the property setter. Correcting it. This will be a no-changelog-needed PR.

Comment (nstarman) [reply astropy/coordinates/angles.py]: Good catch! I missed the property setter. This PR will now be a no-changelog-needed since it's just documenting and adding 2 tests.

**Review by mhvk — APPROVED** [review]
OK, a few extra tests and comments are good to have. Merging...
